src/platform/w3dengine/client/w3dpropbuffer.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Prop Drawing Code
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "asciistring.h"
+#include "partitionmanager.h"
+#include "sphere.h"
+#include "xfer.h"
+
+class CameraClass;
+class LightClass;
+class RenderInfoClass;
+class RenderObjClass;
+class W3DShroudMaterialPassClass;
+
+struct TPropType
+{
+    RenderObjClass *render_obj;
+    Utf8String name;
+    SphereClass bounding_sphere;
+};
+
+struct TProp
+{
+    RenderObjClass *render_obj;
+    int id;
+    Coord3D position;
+    int prop_type;
+    ObjectShroudStatus shroud_status;
+    bool is_visible;
+    SphereClass bounding_sphere;
+};
+
+class W3DPropBuffer : public SnapShot
+{
+public:
+    W3DPropBuffer();
+    ~W3DPropBuffer();
+
+    virtual void CRC_Snapshot(Xfer *xfer) override {}
+
+    virtual void Xfer_Snapshot(Xfer *xfer) override
+    {
+        unsigned char version = 1;
+        xfer->xferVersion(&version, 1);
+    }
+
+    virtual void Load_Post_Process() override {}
+
+    void Cull(CameraClass *camera);
```

Comment:
missing W3DPropBuffer::updateCenter

Reply:
I found no evidence to suggest that it was called updateCenter and gave it the name redoCull instead since it sets field m_redoCull to true. If you have evidence to call it updateCenter I will rename it.